simple-http.js

```diff
@@ -0,0 +1,146 @@
+
+    } else if (auth) 
+    {
+        var tmp = auth.split(' '); // split on space to grab everything after Basic in the auth header
+        var buf = new Buffer(tmp[1], 'base64');
+        
+        // decode base64
+        var plainAuth = buf.toString();
+
+        // basic auth is structured like this: username:password
+        var creds = plainAuth.split(':');
+
+        var username = creds[0];
+        var password = creds[1];
+
+        // uncomment below to see what the hashed username and password are
+        // this is only here to easily see a new username and password to test against below
+        // console.log('hashed username: ', md5(username));
+        // console.log('hashed password: ', md5(password));
+
+        console.log('secret username: ', secrets.username);
+        console.log('secret password: ', secrets.password);
+
+        if((username = secrets.username) && (password == secrets.password))
+        {
+            res.statusCode = 200;
+            res.end('<html>' +
+                        '<body>' +
+                            '<h1>The diamonds are behind the painting</h1>'+
+                        '</body>' +
+                    '</html>');
+        } else {
+            res.statusCode = 401;
+            res.setHeader('WWW-Authenticate', 'Basic realm="Secure Area"');
+
+            res.end('<html>' +
+                        '<body>' +
+                            '<h1>You dont seem to be on the list, let me ask my manager</h1>'+
+                        '</body>' +
+                    '</html>');
+        }
+    }
+});
+
+///
+/// listen call
+/// identical to net server.listen, also creates a listener for the events below
+///
+server.listen(5000, function() 
+{
+    console.log("Middleman listening on http://localhost:5000/");
+});
+
+///
+/// main middleman logic
+/// create client from existing connection, forward data to final destination
+///
+server.on('connection', function(socket) 
+{
+    console.log('A new connection has been established.');
+  
+    // Now that a TCP connection has been established, the server can send data to
+    // the client by writing to its socket.
+    socket.write('Hello, source client.');
+  
+    // The server can also receive data from the client by reading from its socket.
+    socket.on('data', function(chunk) 
+    {
+        console.log(`Data received from client: ${chunk.toString()}`);
+    
+        // BEGIN ACTING AS CLIENT
+        // fwd data to final server
+        var client = new net.Socket();
+  
+        // TODO: Where should this forward to?
+        //       Localhost control room port?
+        client.connect(forwardPort, '127.0.0.1', function() 
+        {
+            console.log('Connected to final server');
+            client.write(chunk.toString());
+        });
+    
+        client.on('data', function(data) 
+        {
+            console.log('Received: ' + data);
+            client.destroy(); // kill client after server's response
```

Comment:
do we need to destroy the client each time?

Reply:
Left a large TODO to determine if this was ever needed once we have the control room set up on the control room PC